rocketpool/node/collectors/node-collector.go

```diff
@@ -339,6 +352,33 @@ func (collector *NodeCollector) Collect(channel chan<- prometheus.Metric) {
 		return nil
 	})
 
+	// get the beacon client sync status:
+	wg.Go(func() error {
+		progress := float64(0)
+		
+		syncStatus, err := collector.bc.GetSyncStatus()
+
+		if err != nil {
+			// don't return an error because then the metric isn't emitted at all, just let the Progress be 0
```

Comment:
I worry that this will be very noisy on startup.

If rocketpool_node and the beacon node are restarted, then prometheus restarts milliseconds later, it will scrape rocketpool_node potentially before the beacon node is accepting connections, resulting in a timeout or tcp or http error here.

I don't think we care to alert on sync status when there is an error, and omitting the metric is probably best. Alerting when the beacon node is up, replying, and sync is <1 is much more _specific_, while doing it the way you've handled it here is much more _sensitive_.

I think specific wins out in this case because i anticipate other alerts to fire if the bn is offline

Reply:
I'm open here. Just for some context:

I've tested this (and just did again) by shutting down the eth2 container entirely and scraping the metrics endpoint rapidly and all works fine. There is now **three** log messages instead of two on _each scrape_ but that's all (the others are `[Node Collector] Error getting beacon chain head: no Beacon clients were ready` and `[Beacon Collector]`). I could remove this log message?

If you're worried about noisy alerts we can control that in the prometheus rule configuration for the alert so that it won't fire unless it's offline for a specified period. 

So with that in mind, what do you prefer to be done here specifically?

Reply:
log messages are not a concern (unless we're talking a dozen). If the alertmanager `for` configs make restarts sufficiently quiet, I don't mind it as-is.